### PR TITLE
Add ESModule tracking to StallMonitor

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- added signal postESMOduleRegistrationSignal_ to ActivityMonitor
- extended StallMonitor to track ESModules. These show up in the global transitions.

#### PR validation:

Code compiles. The extended unit tests succeeds. The stall file generated by the unit test was passed to the graph building script and the generated file looked correct.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Services/plugins/StallMonitor.cc [hunk 1]
```diff
@@ -394,27 +434,47 @@ void StallMonitor::postBeginJob() {
...
+      oss << '\n';
+      file_.write(oss.str());
     }
-    oss << '\n';
-    file_.write(oss.str());
   }
   // Don't need the labels anymore--info. is now part of the
   // module-statistics objects.
   moduleLabels_.clear();
+  esModuleLabels_.clear();
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply FWCore/Services/plugins/StallMonitor.cc:477]: yes, just following what was done in `moduleLabels_`.

[the remark this answers is not in the document]
Comment (makortel) [reply FWCore/Services/plugins/StallMonitor.cc:477]: How about something along
```suggestion
  std::swap(esModuleLabels_, std::vector<std::string>());
```
(for `moduleLabels_` too) to release the reserved memory of the vectors too?

[the remark this answers is not in the document]
Comment (Dr15Jones) [reply FWCore/Services/plugins/StallMonitor.cc]: done

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
All the test failures appear to be from the IB having multiple instances of the same class in multiple libraries which causes ROOT to throw our exception.

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+1

Unit test failure looks like a timeout (and unrelated, and I've seen it fail sporadically in other PRs too)

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**